src/BarChart/BarChart.tsx

```diff
@@ -244,11 +237,9 @@ export class BarChart extends Component<BarChartProps> {
       groupScale,
       keyScale
     };
-  }
-
-  getKeyScale(data, axis, isMultiSeries: boolean, width: number) {
-    const { series } = this.props;
+  };
 
+  const getKeyScale = (data, axis, isMultiSeries: boolean, width: number) => {
```

Comment:
I need to move the `ChartContainer` to be a hook and then a lot of this code could just be memoized and no need for the `renderChart` method too. Just a side note for later.